app/helpers/application_helper.rb

```diff
@@ -13,4 +13,10 @@ def hash_array_json2csv(input_file, output_file, keys)
       end
     end
   end
+
+  def escape_json(hash)
+    str = hash.to_json unless hash.class == 'String'
+    str = str.gsub!("\\", "\\\\\\") if str.include? "\\"
+    str
+  end
```

Comment:
nice, as we discussed. Upon further reflection, there must be a built-in, more secure way to pass the data payload to react_component() without this type of manual escape. I've pinged Trevor for help, hopefully we can discuss soon

Reply:
Also @yunfangjuan ?